### PR TITLE
ClangImporter: expand `-sdk` appropriately on Windows

### PR DESCRIPTION
**Pull request by compnerd**
The MSVC environment does not have a `-isysroot` or `-sysroot`
equivalent.  Instead, manually expand the sysroot to the path as a
prefix to `/usr/include` and treat the path as a system include
directory.  This fixes a number of tests which use `-sdk` on Windows.

Replace this paragraph with a description of your changes and rationale. Provide links to external references/discussions if appropriate.

Resolves [SR-NNNN](https://bugs.swift.org/browse/SR-NNNN).

### CODE REVIEW

**Comment by compnerd** [comment]
@swift-ci please test

File: lib/ClangImporter/ClangImporter.cpp [hunk 1]
```diff
@@ -563,14 +563,21 @@ getNormalInvocationArguments(std::vector<std::string> &invocationArgStrs,
     invocationArgStrs.push_back("-Xclang");
     invocationArgStrs.push_back("-nostdsysteminc");
   } else {
-    // On Darwin, Clang uses -isysroot to specify the include
-    // system root. On other targets, it seems to use --sysroot.
-    if (triple.isOSDarwin()) {
-      invocationArgStrs.push_back("-isysroot");
+    if (triple.isWindowsMSVCEnvironment()) {
+      llvm::SmallString<261> path;
```
Comment (jrose-apple) [inline lib/ClangImporter/ClangImporter.cpp]: Where did this oddly specific number come from?

Comment (xwu) [reply lib/ClangImporter/ClangImporter.cpp]: In the Windows API (with some exceptions), [the maximum length for a path is MAX_PATH, which is defined as 260 characters.](https://msdn.microsoft.com/en-us/library/windows/desktop/aa365247(v=vs.85).aspx)

**Comment by compnerd** [comment]
@swift-ci please test

File: lib/ClangImporter/ClangImporter.cpp (hunk 1 above)
Comment (compnerd) [reply lib/ClangImporter/ClangImporter.cpp]: 261 is `MAX_PATH + 1`, which is pretty much the default size used for path buffers on Windows (the +1 is to ensure null-termination).

Comment (jrose-apple) [reply lib/ClangImporter/ClangImporter.cpp]: Can you put that in instead, then?

Comment (compnerd) [reply lib/ClangImporter/ClangImporter.cpp]: Sure

**Comment by compnerd** [comment]
@swift-ci please test

**Comment by compnerd** [comment]
@jrose-apple ugh, we can't do that because `MAX_PATH` is defined in `Windows.h`  Do you want me to define it locally?

**Comment by compnerd** [comment]
@swift-ci please test

**Comment by jrose-apple** [comment]
Oh, right, cross-compilation is a thing, so even conditionally-including Windows.h would be wrong. Okay, a comment is fine.

**Comment by compnerd** [comment]
@swift-ci please test and merge

**Comment by compnerd** [comment]
@swift-ci please test and merge

**Pull request merged**